Quality of life improvements.

ncareau opened the pull request:
I propose a few changes that simplify and I think help the project.

- When using no contract, use the first available and warn user. (Before, when no contract specified, the app crashed.)
- Why not use only one dockerfile ? I've added the mqtt output in the same dockerfile and moved to the root. In the long run, I think having only one docker will be easier for users. 
- Added a way to use environment variables instead of cli. This is the way to go for docker/k8s as environment are the principal way to configure an app.
- Added .vscode to gitignore.

I'm open to changes, deletion or addition. Just let me know what you think. 

I think that the docker changes will need changes to your .gitlab-ci file. Are you planning to move to gitlab ? or just use it for the registry ?

-NMC

titilambert commented:
BIG THANKS
This is a huge one !

titilambert commented on pyhydroquebec/__main__.py:
```diff
@@ -96,22 +100,42 @@ def main():
+    hydro_user = hydro_pass = hydro_contract = None
+    #Check Env
+    if "PYHQ_USER" in os.environ:
+        hydro_user = os.environ["PYHQ_USER"]
```
Can be replaced by `hydro_user = os.environ.get("PYHQ_USER")`
If the key is not there, the value will be None.

ncareau replied:
Done

titilambert commented on pyhydroquebec/__main__.py:
```diff
@@ -96,22 +100,42 @@ def main():
 
-    if not args.username or not args.password:
+    # Check input for Username, Password and Contract - CLI overwrite ENV variable
+    hydro_user = hydro_pass = hydro_contract = None
```
If you use `os.environ.get...` things this line can be removed

ncareau replied:
Done

titilambert commented on pyhydroquebec/__main__.py:
```diff
@@ -96,22 +100,42 @@ def main():
+        hydro_user = os.environ["PYHQ_USER"]
+
+    if "PYHQ_PASSWORD" in os.environ:
+        hydro_pass = os.environ["PYHQ_PASSWORD"]
```
Same here

ncareau replied:
Done

titilambert commented on pyhydroquebec/__main__.py:
```diff
@@ -96,22 +100,42 @@ def main():
+        hydro_pass = os.environ["PYHQ_PASSWORD"]
+
+    if "PYHQ_CONTRACT" in os.environ:
+        hydro_contract = os.environ["PYHQ_CONTRACT"]
```
Same here

ncareau replied:
Done

titilambert commented on pyhydroquebec/__main__.py:
```diff
@@ -96,22 +100,42 @@ def main():
         parser.print_usage()
         print("pyhydroquebec: error: the following arguments are required: "
-              "-u/--username, -p/--password")
+                "-u/--username, -p/--password")
```
Keep it aligned.
The `tox -e pylint,lint` command should highlight this issue

ncareau replied:
Done

ncareau commented:
Updated the PR. 

I've removed the vscode line in gitignore, you are right that this should be in my system gitignore file.

Also note that github has a registry (in beta) and multiple free CI tools that works well with it (ie. travix-ci). You can even plug the lint step in it to block PR from merging if the lint returns false.

There is also using docker hub (which is currently the defacto for docker container repository) which has automatic build of images. You can probably setup something in less than 10 min without the need for script (ie, the gitlab-ci.yml file) as it uses the Dockerfile directly. Let me know if you need help setting that up. 

Even after all that gitlab is a great option too, i use it a lot for all my private projects.
